db/bigtable_eth1.go

```diff
@@ -2542,11 +2542,29 @@ func (bigtable *Bigtable) GetInternalTransfersForTransaction(transaction []byte,
 		from := utils.FormatAddress(t.From, nil, fromName, false, false, true)
 		to := utils.FormatAddress(t.To, nil, toName, false, false, true)
 
-		data[i] = types.Transfer{
-			From:   from,
-			To:     to,
-			Amount: utils.FormatBytesAmount(t.Value, utils.Config.Frontend.ElCurrency, 8),
+		input := make([]byte, 0)
+		if len(parityTrace[i+1].Action.Input) > 2 {
+			input, err = hex.DecodeString(parityTrace[i+1].Action.Input[2:])
+			if err != nil {
+				utils.LogError(err, "can't convert hex string", 0)
+			}
 		}
+
+		data[i] = types.ITransaction{
+			From:      from,
+			To:        to,
+			Amount:    utils.FormatBytesAmount(t.Value, utils.Config.Frontend.ElCurrency, 8),
+			TracePath: utils.FormatTracePath(parityTrace[i+1].Action.CallType, parityTrace[i+1].TraceAddress, parityTrace[i+1].Error == "", bigtable.GetMethodLabel(input[:4], true)),
+		}
+
+		gaslimit, err := strconv.ParseUint(parityTrace[i+1].Action.Gas, 0, 0)
+		if err == nil {
+			data[i].Gas.Limit = gaslimit
+		}
```

Comment:
**issue:** I couldn't test this for failed transactions because I couldn't find a tx on Prater that fulfilled the requirements.

However I could check normal tx with itx and the resulting table is not correct.
Take e.g. tx
https://goerli.etherscan.io/tx/0x49359f1d79e8c44bb7ce859e558b28c74cbb6df02b173fda0cb7510afebd75a6#internal

It has 5 itx but the only one we would want to show is the last one (the call), the same one you see in the Overview tab.
If I start the branch locally and check /tx/0x49359f1d79e8c44bb7ce859e558b28c74cbb6df02b173fda0cb7510afebd75a6#internal-txns
I get one line.
The From, To and Value that we already had on master are correct but the new Type Trace Address and Gas Limit are not.
![Screenshot from 2024-01-05 15-36-24](https://github.com/gobitfly/eth2-beaconchain-explorer/assets/124689083/3ed8a9c7-93ee-458e-a3b9-f9c8e7a65ed3)
They are taken from the first **delegatecall** line.

The order of `parityTrace` we have is:
0. Base "call" from 0x734D53...2EF3d7e4 to 0x28D8F1...4E75175a
1. delegatecall_0_1
2. staticcall_0_1_1
3. delegatecall_0_1_1_1
4. call_0_1_1
(No idea what happened with staticcall_0_1_1_1_1)

I assume we need to work more with `t`/`k` and not `i` but please have a look.

Reply:
Thanks for your feedback. Mb, got confused with the follow up PR which would've taken care of this; that's included here now. Lmk if it works now

Reply:
Ok, that nearly works as I would expect it to, see my other comment regarding `Advanced`.